content/blog/2024/01/2024-01-04-migrating-from-ems-to-selfhosted-matrix.md

```diff
@@ -0,0 +1,224 @@
+2. After all users have completed those steps, cut over your DNS and .well-known files. This stops traffic from being directed to your EMS host.
+3. Wait twice as long as your old DNS TTL values were set to, to allow for propagation with a fudge factor.
+4. Request a backup from EMS. It took about 2 hours from the time I requested my export for the database export to be ready, and 2.5 hours for the media repo export to be ready.
+5. Import key configuration values, do initial setup of your homeserver, and import your database. Do not start Synapse until the database is imported! **The “set up your homeserver” instructions below are primarily focused on this step.**
+    1. If this goes badly wrong, restore DNS and fix.
+6. Turn your Synapse on. It'll roar to life and eat a bunch of CPU as it catches up, but that's fine.
+7. Wait for things to catch up. In this time, ask your key contacts to re-send any messages they may have sent while you've been offline.
+8. Also while waiting, import your media. The instructions below also cover this step.
+9. If things look good after about 2-4 hours, cancel and delete your EMS homeserver (it'll now be too far behind to roll back to).
+10. Rotate your signing key at your convenience.
+
+## Set up your homeserver
+
+These instructions are focused on setting up your homeserver, importing your database, and importing your media. Here are some assumptions I’m making in what I’ve written:
+
+* Dependencies are already addressed locally and remotely.
+* You don’t have any bridges setup on EMS. (Their [docs](https://ems-docs.element.io/books/element-cloud-documentation/page/migrate-from-ems-to-self-hosted) cover what to do if you do have bridges.)
+* You’ve cloned the MDAD project locally and configured your playbook.
+  * You’re using the default profile (Synapse, Element, Postgres, Coturn, Traefik, Let’s Encrypt, and Exim) and, of the optional services, have opted to use matrix-media-repo (MMR).
+* You’ve already set your target host in the MDAD project.
+* You’ve already put the `macaroon` and `pepper` keys from your EMS export into the MDAD project’s `vars.yml`, under `matrix_synapse_macaroon_secret_key` and `matrix_synapse_password_config_pepper` respectively.
+* You’re using a bare domain for your account handles (as in, not using a subdomain).
+* You’ve already handled the SSL certificates and prepared your DNS.
+* You’ll store your EMS exports locally at `~/dev/migration`.
+* You’ve got a directory setup on the host server at `/root/migration` – and you run commands on the host as root.
+
+OK, with all that scene setting done, let’s walk through the steps you’ll need to take to set up your homeserver and import your data!
+
+### Run your Ansible playbook
+
+From your local machine, run the Ansible playbook. All of the commands we run from the local machine are within the base directory of the MDAD project. Critically, we’re only installing things – not starting any services yet.
+
+```
+> just roles
```

Comment:
I might be out of date with the playbook/ansible, but is this a command and if so where is it from? :)

Reply:
Specifically the `just` command

Reply:
Got that from p2: https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/docs/installing.md

Thank you for the detailed review, the more eyes on this the better I feel about it! 😅

Reply:
Ah ok. Maybe it makes sense to just link/introduce it once here in the post to keep it flowing :) The ansible commands are clear considering its an ansible playbook. :(